test/content-routing.node.js

```diff
@@ -7,87 +7,363 @@ const chai = require('chai')
-          EXPERIMENTAL: {
-            dht: true
+  describe('via the dht', () => {
+    let nodeA
+    let nodeB
+    let nodeC
+    let nodeD
+    let nodeE
+
+    before(function (done) {
+      this.timeout(5 * 1000)
+      const tasks = _times(5, () => (cb) => {
+        createNode('/ip4/0.0.0.0/tcp/0', {
+          config: {
+            EXPERIMENTAL: {
+              dht: true
+            }
           }
-        }
-      }, (err, node) => {
+        }, (err, node) => {
+          expect(err).to.not.exist()
+          node.start((err) => cb(err, node))
+        })
+      })
+
+      parallel(tasks, (err, nodes) => {
         expect(err).to.not.exist()
-        node.start((err) => cb(err, node))
+        nodeA = nodes[0]
+        nodeB = nodes[1]
+        nodeC = nodes[2]
+        nodeD = nodes[3]
+        nodeE = nodes[4]
+
+        parallel([
+          (cb) => nodeA.dial(nodeB.peerInfo, cb),
+          (cb) => nodeB.dial(nodeC.peerInfo, cb),
+          (cb) => nodeC.dial(nodeD.peerInfo, cb),
+          (cb) => nodeD.dial(nodeE.peerInfo, cb),
+          (cb) => nodeE.dial(nodeA.peerInfo, cb)
+        ], done)
       })
     })
 
-    parallel(tasks, (err, nodes) => {
-      expect(err).to.not.exist()
-      nodeA = nodes[0]
-      nodeB = nodes[1]
-      nodeC = nodes[2]
-      nodeD = nodes[3]
-      nodeE = nodes[4]
-
+    after((done) => {
       parallel([
-        (cb) => nodeA.dial(nodeB.peerInfo, cb),
-        (cb) => nodeB.dial(nodeC.peerInfo, cb),
-        (cb) => nodeC.dial(nodeD.peerInfo, cb),
-        (cb) => nodeD.dial(nodeE.peerInfo, cb),
-        (cb) => nodeE.dial(nodeA.peerInfo, cb)
+        (cb) => nodeA.stop(cb),
+        (cb) => nodeB.stop(cb),
+        (cb) => nodeC.stop(cb),
+        (cb) => nodeD.stop(cb),
+        (cb) => nodeE.stop(cb)
       ], done)
     })
-  })
 
-  after((done) => {
-    parallel([
-      (cb) => nodeA.stop(cb),
-      (cb) => nodeB.stop(cb),
-      (cb) => nodeC.stop(cb),
-      (cb) => nodeD.stop(cb),
-      (cb) => nodeE.stop(cb)
-    ], done)
+    it('should use the nodes dht to provide', (done) => {
+      const stub = sinon.stub(nodeA._dht, 'provide').callsFake(() => {
+        stub.restore()
+        done()
+      })
+
+      nodeA.contentRouting.provide()
+    })
+
+    it('should use the nodes dht to find providers', (done) => {
+      const stub = sinon.stub(nodeA._dht, 'findProviders').callsFake(() => {
+        stub.restore()
+        done()
+      })
+
+      nodeA.contentRouting.findProviders()
+    })
+
+    describe('le ring', () => {
+      const cid = new CID('QmTp9VkYvnHyrqKQuFPiuZkiX9gPcqj6x5LJ1rmWuSySnL')
+
+      it('let kbucket get filled', (done) => {
+        setTimeout(() => done(), 250)
+      })
+
+      it('nodeA.contentRouting.provide', (done) => {
+        nodeA.contentRouting.provide(cid, done)
+      })
+
+      it('nodeE.contentRouting.findProviders for existing record', (done) => {
+        nodeE.contentRouting.findProviders(cid, 5000, (err, providers) => {
```

Comment:
Use the`maxTimeout` instead of `5000`, as we intend to deprecate it in the future.